ci/it/testcases/test_ingest.go

```diff
@@ -645,3 +654,423 @@ func (a *IngestTestcase) testFieldEncodingsMappingsBug(ctx context.Context, t *t
 	assert.Equal(t, "quesmaMetadataV1:fieldName=Field1", comments["field1"])
 	assert.Equal(t, "quesmaMetadataV1:fieldName=Field2", comments["field2"])
 }
+
+// Struct to parse only the `fields` tree
+type Hit struct {
+	Fields map[string][]any `json:"fields"`
+	Source map[string]any   `json:"_source"`
+}
+
+type HitsWrapper struct {
+	Hits []Hit `json:"hits"`
+}
+
+type Response struct {
+	Hits HitsWrapper `json:"hits"`
+}
+
+func ParseResponse(t *testing.T, body []byte) map[string]any {
+	var response Response
+	err := json.Unmarshal([]byte(body), &response)
+	if err != nil {
+		log.Fatalf("Error parsing JSON: %v", err)
+	}
+
+	// Extract and print the `fields` tree
+	for _, hit := range response.Hits.Hits {
+		return hit.Source
+	}
+	return nil
+}
```

Comment:
Maybe move this code to some other file? Model, utils, or some other name?

When some (useful for other tests) classes are declared in the middle of a very long test file, it may be hard to find and reuse them